compiler/rustc_middle/src/ty/vtable.rs

```diff
@@ -45,6 +47,65 @@ pub const COMMON_VTABLE_ENTRIES_DROPINPLACE: usize = 0;
 pub const COMMON_VTABLE_ENTRIES_SIZE: usize = 1;
 pub const COMMON_VTABLE_ENTRIES_ALIGN: usize = 2;
 
+// FIXME: This is duplicating equivalent code in compiler/rustc_trait_selection/src/traits/util.rs
+// But that is a downstream crate, and this code is pretty simple. Probably OK for now.
```

Comment:
Why can't `rustc_trait_selection` invoke this code instead?

Like you could add a `TyCtxt` inherent method here that returns an `impl Iterator<Item = DefId>`

Reply:
Good point, did that in an extra commit.